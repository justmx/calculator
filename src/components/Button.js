import React from 'react';

const Button = ({content, onClick, left}) => {	
	const containStyle = left ? ['circle-holder-left'] : ['circle-holder-right'];
	const style = ['circle'];
	if(content.double){
		style.push('double');
		containStyle.push('double')
	}

	return (
		<div onClick={() => onClick(content.content)} className={containStyle.join(' ')} >
			<div className={style.join(' ')} 
				style={{background: content.colour || '#363636'}}>{content.content}</div>
		</div>
	)
}

export default Button;