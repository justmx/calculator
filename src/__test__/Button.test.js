import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button';

const clickFn = jest.fn();
const content = {content: '1', colour: '#ffffff'};
const contentZero = {
  content: '0',
  colour: '#363636',
  double: true
}

describe('Button', () => {
  it('should render content', () => {
    const wrapper = shallow(<Button content={content} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render button with a double class', () => {
    const wrapper = shallow(<Button content={contentZero} />);
    expect(wrapper.find('.circle').hasClass('double')).toEqual(true);
  });

  it('should fire a function when click', () => {
	  const wrapper = shallow(<Button content={content} onClick={clickFn} />);
	  wrapper
	    .find('.circle-holder-right')
	    .simulate('click');
	 expect(clickFn).toHaveBeenCalled();
  });
});