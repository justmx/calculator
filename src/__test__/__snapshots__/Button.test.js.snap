// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render content 1`] = `
<div
  className="circle-holder-right"
  data-testid="1"
  onClick={[Function]}
>
  <div
    className="circle"
    style={
      Object {
        "background": "#ffffff",
      }
    }
  >
    1
  </div>
</div>
`;
